src/alignment/segments.py

```diff
@@ -41,30 +42,87 @@ def checkForConflicts(self, other: AlignmentSegment):
-            lambda p: not isinstance(p, AlignedPair) or p.lessOnBothSequences(start), self.positions)
+            lambda p: p.lessOnBothSequences(start), self.positions)
         positions = list(
             itertools.takewhile(
                 lambda p: not isinstance(p, AlignedPair) or p.lessOrEqualOnAnySequence(end),
                 slicedAtStart))
-        self.__trimNotAlignedPositionsFromEnd(positions)
+        self.__trimNotAlignedPositionsFromEnd(positions, end)
         return AlignmentSegment(positions, sum(p.score for p in positions), self.peak, self.allPeakPositions)
 
+    def get_reference_labels(self) -> Tuple[List, List, List]:
+        """Function used to get all of the reference labels,
+        scores and their indexes present in a segment
+
+        :return: Characteristics of Reference in a segment of alignment
+        :rtype: Tuple[List, List, List]
+        """
+        ref_pos, ref_scores, ref_index = [], [], []
+        sum_score = 0
+        for index, position in enumerate(self.positions):
+            if isinstance(position, ScoredAlignedPair):
+                ref_index.append(index)
+                ref_pos.append(position.reference)
+                ref_scores.append(position.score + sum_score)
+                sum_score = 0
+            else:
+                if not isinstance(position.position, NotAlignedQueryPosition):
+                    ref_index.append(index)
+                    ref_pos.append(position.position.reference)
+                    ref_scores.append(position.score + sum_score)
+                    sum_score = 0
+                else:
+                    sum_score += position.score
+        return (ref_pos, ref_scores, ref_index)
+
+    def get_query_labels(self) -> Tuple(List, List, List):
+        """Function used to get all of the query labels,
+        scores and their indexes present in a segment
+
+        :return: Characteristics of Query in a segment of alignment
+        :rtype: Tuple[List, List, List]
+        """
+        quer_pos, quer_scores, quer_index = [], [], []
+        sum_score = 0
+        for index, position in enumerate(self.positions):
+            if isinstance(position, ScoredAlignedPair):
+                quer_pos.append(position.query)
+                quer_scores.append(position.score + sum_score)
+                quer_index.append(index)
+                sum_score = 0
+            else:
+                if isinstance(position.position, NotAlignedQueryPosition):
+                    quer_pos.append(position.position.query)
+                    quer_scores.append(position.score + sum_score)
+                    quer_index.append(index)
+                    sum_score = 0
+                else:
+                    sum_score += position.score
+        return (quer_pos, quer_scores, quer_index)
+
     @staticmethod
-    def __trimNotAlignedPositionsFromEnd(positions):
+    def __trimNotAlignedPositionsFromEnd(positions, end=None):
```

Comment:
Otherwise this function drops unpaired labels from the end where the one that is present corresponds to the same one (query or reference) in the “end” argument used in slice function. Those labels are needed while looking for the optimal place to slice conflicting segments. That’s why I have chosen to default it to None, as to keep the option to also trim those labels. Per your request I have added a test in “test_segments.py”- test_slice_parameterEnd_keepsUnpaired